docs/proposal/machine-config.md

```diff
@@ -0,0 +1,65 @@
+# AWS Provider Machine Configuration and Status
+
+## Overview
+This proposal aims to provide an initial API for AWS machine configuration and status for the AWS machine actuator.
+Cluster configuration for the AWS provider is out of scope for this proposal.
+
+## AWSMachineProviderConfig
+Defines the spec portion of a Machine specific to AWS. It is meant to be embedded in the
```

Comment:
Are we focusing this API only for on-demand AWS instances? If we want to have spot instances support then we might want to include additional fields.

Reply:
I would say that the we could initially focus on on-demand (because it's simpler), but we could eventually evolve to support spot instances.